lib/dndtable.sty

```diff
@@ -9,11 +9,31 @@
 \newcolumntype{Y}{>{\centering\arraybackslash}X}
 
 % Table Environment
-\newenvironment{dndtable}[1][XX]{
-	\par\vspace*{8pt}
-	\noindent
-	\fontfamily{lmss}\selectfont %Select font
-	\rowcolors{1}{}{commentgreen} % Alternate colors
-	\tabularx{\linewidth}{#1}
-	}
-	{\endtabularx\vspace{8pt plus 1pt}\noindent} 
+
+% \tabularx expands to commands that directly hack the typesetting system. The
+% macro expects and end token, similar to a real environment.
+% Wrapping an environment that uses tabularx causes a runaway argument error
+% because tabularx cannot find the end token.
+% Instead, create two command that expand to the table header and footer.
+\newcommand{\dnd@starttable}[1]{%
+  \par\vspace*{8pt}
+  \noindent
+  \fontfamily{lmss}\selectfont
+  \rowcolors{1}{}{tablecolor}
+  \tabularx{\linewidth}{#1}
+}
+
+\newcommand{\dnd@endtable}{\endtabularx\vspace{8pt plus 1pt}\noindent}
+
+\newenvironment{dndtable}[1][XX]{\dnd@starttable{#1}}{\dnd@endtable}
+
+\newenvironment{dnditemtable}[1][XX]
```

Comment:
dnditemtable is deprecated by this change and should probably not be changed to fit into the new color method of handling colors. If you are feeling adventurous, you could try adding this to the beginning of the original dnditemtable definition:
```
\PackageWarning{dnd}{dnditemtable is deprecated. use dndtable after setting tablecolor instead.}
```
You could also change the definition of the dndtable environment to take an optional color like paperbox does.

Reply:
Done.